firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/DataCollectionArbiter.java

```diff
@@ -107,13 +116,29 @@ public boolean isAutomaticDataCollectionEnabled() {
   }
 
   @SuppressLint({"CommitPrefEdits", "ApplySharedPref"})
-  public void setCrashlyticsDataCollectionEnabled(boolean enabled) {
-    crashlyticsDataCollectionEnabled = enabled;
-    crashlyticsDataCollectionExplicitlySet = true;
-    sharedPreferences.edit().putBoolean(FIREBASE_CRASHLYTICS_COLLECTION_ENABLED, enabled).commit();
+  public synchronized void setCrashlyticsDataCollectionEnabled(Boolean enabled) {
+    if (enabled == null) {
```

Comment:
Question: this setting, conceptually, should be "clear the shared preferences setting, then reset the values based on the manifest". Is there any way we could consolidate that second part with the same conceptual code (i.e. setting the values based on the manifest) in the constructor, and make a method out of it? That would 1) make this method simpler and more compact, and 2) make sure that both times (on construction and also here) we set the values the same way.